File: regex/CSharp/regex4.cs
using System;
using System.Text.RegularExpressions;

class Regex4
{
  public static String myEval( Match mat )
  {
    if ( mat.Groups[ "key" ].ToString() == "name" )
      return "Jack";

    if ( mat.Groups[ "key" ].ToString() == "cost" )
      return "$20";

    return "";
  }

  static void Main()
  {
    Regex r = new Regex( "<(?<key>.*?)>" );
    String str = r.Replace(
                    "Dear <name>, You owe us <cost>",
                    new MatchEvaluator( myEval ) );
    Console.WriteLine( str );
  }
}

File: regex/Ruby/regex4.rb
str = "Dear <name>, You owe us <cost>"

str.gsub!( /<(.*?)>/ ) { 

  if ( $1 == "name" )
	"jack"
  elsif ( $1 == "cost" )
	"$20"
  else
	""
  end

}

print "#{str}\n"
